### DISCUSSION TITLE
connecting user to group

### DISCUSSION THREAD

**Discussion by samisalamiws** [containerbase/base#267] [Q&A] [answered]
Hi @viceice 
before buildpack we were doing:
`RUN groupadd ${WSS_GROUP} && \`
`	useradd --gid ${WSS_GROUP} --groups 0 --shell /bin/bash --home-dir ${WSS_USER_HOME} --create-home ${USER_NAME} && \`
`	passwd -d ${USER_NAME}`

So I had to choose the group name that I want.

now it's:

`groupadd --gid ${USER_ID} ${USER_NAME};`
`useradd --uid ${USER_ID} --gid 0 --groups ${USER_NAME} --shell /bin/bash --create-home ${USER_NAME}`

which sets the group name as the user name. 
plus, the `useradd` command now doing  `--gid 0 --groups ${USER_NAME} ` which connects the user to the root group, why is that so?

**Comment by viceice** [comment, marked as answer]
https://github.com/containerbase/buildpack/blob/main/docs/custom-base-image.md#custom-user-name-and-id ?

**Reply by viceice** [reply]
The `root` group is required, so the user can modify some protected files

**Reply by viceice** [reply]
it's also required for openstack compatibillity

**Reply by samisalamiws** [reply]
yes, I'm setting the USER_NAME and USER_ID as suggested. I was only wondering about the group name (not so important, just checking if there is a way)
thanks

**Reply by viceice** [reply]
Currently not, you can open an issue and provide your use case. Maybe we can add a `GROUP_NAME` variable which falls back to `USER_NAME`.

I'm not sure if we can change the default group.
